datadog_checks_base/tests/test_http.py

```diff
@@ -124,7 +124,7 @@ def test_extra_headers_on_http_method_call(self):
         http = RequestsWrapper(instance, init_config)
 
         extra_headers = {"foo": "bar"}
-        with mock.patch("requests.get") as get:
+        with mock.patch("requests.Session.get") as get:
```

Comment:
Changing all these mock calls took a fair amount of tedious work, but I'm not sure if we should do anything about this.

I thought about parametrizing test cases by the HTTP `method`, since we do _basically_ the same thing for GET, POST, PUT, PATCH, DELETE. But the devil is in the details and maybe we don't exactly do the same thing in each case, though.

Anyway, we can treat any refactor of these big amounts of mocks as a follow-up…

Reply:
Agreed with the idea of pushing that change to a separate PR.